app/bundles/PluginBundle/Bundle/PluginBundleBase.php

```diff
@@ -33,28 +33,41 @@ public static function onPluginInstall(Plugin $plugin, MauticFactory $factory, $
     public static function installPluginSchema(array $metadata, MauticFactory $factory, $installedSchema = null): void
     {
         if (null !== $installedSchema) {
-            // Schema exists so bail
+            // Schema already exists, so no need to proceed
             return;
         }
 
         $db             = $factory->getDatabase();
         $schemaTool     = new SchemaTool($factory->getEntityManager());
         $installQueries = $schemaTool->getCreateSchemaSql($metadata);
 
-        $db->beginTransaction();
-        try {
-            foreach ($installQueries as $q) {
+        foreach ($installQueries as $q) {
+            // Check if the query is a DDL statement
+            if (self::isDDLStatement($q)) {
+                // Execute DDL statements outside of a transaction
                 $db->executeQuery($q);
+            } else {
+                // For non-DDL statements, use transactions
+                try {
+                    $db->beginTransaction();
+                    $db->executeQuery($q);
+                    $db->commit();
+                } catch (\Exception $e) {
+                    // Rollback only for non-DDL statements
+                    if ($db->isTransactionActive()) {
+                        $db->rollBack();
+                    }
+                    throw $e;
+                }
             }
-
-            $db->commit();
-        } catch (\Exception $e) {
-            $db->rollback();
-
-            throw $e;
         }
     }
 
+    private static function isDDLStatement(string $query): bool|int
+    {
+        return preg_match('/^(CREATE|ALTER|DROP)\s/i', $query);
```

Comment:
little nitpick, but there are more keywords than those above.
e.g. TRUNCATE COMMENT and RENAME are also valid.

Reply:
More DDL variants added